api/test/e2e/route_with_plugin_cors_test.go

```diff
@@ -0,0 +1,226 @@
+	tests := []HttpTestCase{
+		{
+			caseDesc:     "make sure the route is not created",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			ExpectStatus: http.StatusNotFound,
+			ExpectBody:   `{"error_msg":"404 Route Not Found"}`,
+		},
+		{
+			caseDesc: "create route with cors default setting",
+			Object:   ManagerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/r1",
+			Body: `{
+				"uri": "/hello",
+				"plugins": {
+					"cors": {}
+				},
+				"upstream": {
+					"type": "roundrobin",
+					"nodes": [{
+						"host": "172.16.238.20",
+						"port": 1981,
+						"weight": 1
+					}]
+				}
+			}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+		{
+			caseDesc:     "verify route with cors default setting",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			ExpectStatus: http.StatusOK,
+			ExpectHeaders: map[string]string{
+				"Access-Control-Allow-Origin":  "*",
+				"Access-Control-Allow-Methods": "*",
+			},
+			ExpectBody: "hello world",
+			Sleep:      sleepTime,
+		},
+		{
+			caseDesc: "update route with specified setting",
+			Object:   ManagerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/r1",
+			Body: `{
+			"uri": "/hello",
+				"plugins": {
+					"cors": {
+						"allow_origins": "http://sub.domain.com,http://sub2.domain.com",
+						"allow_methods": "GET,POST",
+						"allow_headers": "headr1,headr2",
+						"expose_headers": "ex-headr1,ex-headr2",
+						"max_age": 50,
+						"allow_credential": true
+					}
+				},
+				"upstream": {
+					"type": "roundrobin",
+					"nodes": [{
+						"host": "172.16.238.20",
+						"port": 1981,
+						"weight": 1
+					}]
+				}
+			}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+		{
+			caseDesc: "verify route with cors specified setting",
+			Object:   APISIXExpect(t),
+			Method:   http.MethodGet,
+			Path:     "/hello",
+			Headers: map[string]string{
+				"Origin":    "http://sub2.domain.com",
+				"resp-vary": "Via",
+			},
+			ExpectStatus: http.StatusOK,
+			ExpectHeaders: map[string]string{
+				"Access-Control-Allow-Origin":   "http://sub2.domain.com",
+				"Access-Control-Allow-Methods":  "GET,POST",
+				"Access-Control-Allow-Headers":  "headr1,headr2",
+				"Access-Control-Expose-Headers": "ex-headr1,ex-headr2",
+				"Access-Control-Max-Age":        "50",
+			},
+			ExpectBody: "hello world",
+			Sleep:      sleepTime,
+		},
+		{
+			caseDesc: "verify route with cors specified no match origin",
+			Object:   APISIXExpect(t),
+			Method:   http.MethodGet,
+			Path:     "/hello",
+			Headers: map[string]string{
+				"Origin": "http://sub3.domain.com",
+			},
+			ExpectStatus: http.StatusOK,
+			ExpectHeaders: map[string]string{
+				"Access-Control-Allow-Origin":   "",
```

Comment:
It is wrong. If we return all headers with empty or no headers, the browser will block the request.

Ref to :https://developer.mozilla.org/zh-CN/docs/Web/HTTP/Access_control_CORS.

Reply:
updated.